### DISCUSSION TITLE
How to add a custom errorComponent from a module

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#16580] [Questions] [answered]
Hi,

I'd like to add my own error page for all the pages that will include a specific module.

I have the feeling it's possible if I have access to nuxtApp 

    app.errorComponent = resolve(__dirname, './runtime/myerror.vue')

However I don't see how to access to app variable in the module. 
Maybe I have to do that in a hook but, which one ?

**Comment by Diizzayy** [comment, marked as answer]
> Maybe I have to do that in a hook but, which one ?

@hlassiege Hey Hugo, The `app:resolve` hook is what you're looking for, You can leverage it within the module to reference your custom error component as seen below.

```ts
import { createResolver, defineNuxtModule } from '@nuxt/kit'

export default defineNuxtModule({
  setup (options, nuxt) {
    const resolver = createResolver(import.meta.url)

    nuxt.options.build.transpile.push(resolver.resolve('runtime'))

    nuxt.hook('app:resolve', (app) => {
      app.errorComponent = resolver.resolve('runtime/err.vue')
    })
  }
})
```

**Comment by hlassiege** [comment]
Thanks a lot, it works. Just one question, I used 

        nuxt.hook('app:resolve', (app) => {
            app.errorComponent = resolve(__dirname, './runtime/error.vue');
        });

What is the difference with 

    const resolver = createResolver(import.meta.url)
    nuxt.options.build.transpile.push(resolver.resolve('runtime'))
    nuxt.hook('app:resolve', (app) => {
      app.errorComponent = resolver.resolve('runtime/err.vue')
    })

resolve is already available in the context. It is better to create your own resolver ?

**Reply by Diizzayy** [reply]
>  It is better to create your own resolver ?

Not quite and there's little to no difference, it comes down to usage preference. Though I do recommended having the runtime dir transpiled.
